src/test/kotlin/com/charleskorn/kaml/testobjects/TestObjects.kt

```diff
@@ -63,3 +79,126 @@ val sealedModule = SerializersModule {
         TestSealedStructure.SimpleSealedString::class with TestSealedStructure.SimpleSealedString.serializer()
     }
 }
+
+interface SimpleInterface
+
+class SimpleNull : SimpleInterface {
+    companion object {
+        val kSerializer: KSerializer<SimpleNull> = UnitSerializer.nullable.mapped("simpleNull", { SimpleNull() }, { null })
+    }
+
+    override fun equals(other: Any?): Boolean = other is SimpleNull
```

Comment:
Are `equals` and `hashCode` required here? Could we use an `object` instead of a `class` perhaps?